packages/headless/src/features/pagination/pagination-slice.ts

```diff
@@ -70,6 +74,18 @@ export const paginationReducer = createReducer(
       .addCase(executeSearch.fulfilled, (state, action) => {
         const {response} = action.payload;
         state.totalCountFiltered = response.totalCountFiltered;
+      })
+      .addCase(toggleSelectFacetValue, (state) => {
+        state.firstResult = 0;
+      })
+      .addCase(toggleSelectCategoryFacetValue, (state) => {
+        state.firstResult = 0;
+      })
+      .addCase(toggleSelectDateFacetValue, (state) => {
+        state.firstResult = 0;
+      })
+      .addCase(toggleSelectNumericFacetValue, (state) => {
```

Comment:
There is a bit more actions which modify filters. If you look in the facet-set-slice, you'll see it also handles:
- deselectAllFacetValues
- selectFacetSearchResult
- deselectAllFacets
There are more actions for other facet types as well

Reply:
I will add those as well.